app/models/concerns/servers/command.rb

```diff
@@ -98,4 +95,68 @@ def execute_remote run, script_path
         output: out.to_s + status_text.to_s
       }
     end
+
+    def local_execution execution, script_path
+      status = 1
+
+      Open3.popen2e rails, 'runner', script_path do |stdin, stdout, thread|
+        execution.update! pid: thread.pid
+
+        stdout.each { |line| execution.new_line line }
+
+        status = thread.value.exitstatus.to_i
+      end
+
+      status
+    end
+
+    def remote_execution execution, script_path
+      status = 1
+
+      Net::SSH.start hostname, user, ssh_options  do |ssh|
+        # execution permission
+        ssh.exec! "chmod +x #{script_path}"
+
+        output_proc = proc { |data| execution.new_line data }
+
+        status = ssh_exec_with_realtime_output ssh, "$SHELL -c #{script_path}", output_proc
+
+        # Clean the script
+        ssh.exec! "rm #{script_path}"
+      end
+
+      status
+    end
+
+    def ssh_exec_with_realtime_output ssh, command, output_proc
+      status = 1
+
+      channel = ssh.open_channel do |och|
+        # Realtime log output
+        och.request_pty
+
+        # Command execution
+        och.exec command do |ch, success|
+          # STDOUT
+          ch.on_data do |_ch, data|
+            output_proc.call data
```

Comment:
```suggestion
            output_proc[data]
```

De nuevo, se que lo amás en silencio =P. De hecho, usándolo así podría quedar _oneliner_ el `on_data`.

Reply:
![image](https://user-images.githubusercontent.com/873323/61801913-f9c4bb00-ae05-11e9-9dd4-e5a19721e585.png)